docs/query_engine/config/index.md

```diff
@@ -0,0 +1,40 @@
+# Configuration
+
+## Default query engine
+
+By default M3 runs two query engines:
+- Prometheus (default) - robust and de-facto query language for metrics
+- M3 Query Engine - high-performance query engine but doesn't support all the functions yet
+
+Prometheus Query Engine (or PromQL) is the default one when calling query endpoint:
+```
+http://localhost:7201/api/v1/query?query=count(http_requests)&time=1590147165
+```
+
+But you can switch between the two in the following ways:
+- Changing default query engine in config file (see `defaultEngine` parameter in [Configuration](config/annotated_config/))
+- Passing HTTP header `X-M3-Engine` when calling http url
```

Comment:
Can we use just M3- prefix like our other headers? X- is a deprecated prefix as per:
https://tools.ietf.org/html/rfc6648